.github/workflows/ci.yml

```diff
@@ -279,19 +279,25 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # we need the tags to be available
+
       - name: Install
         uses: ./.github/actions/prepare-install
         with:
           node-version: ${{ env.PRIMARY_NODE_VERSION }}
           registry-url: 'https://registry.npmjs.org'
+
       - name: Build
         uses: ./.github/actions/prepare-build
 
-      # Fetch all history for all tags and branches in this job because lerna needs it
-      - run: |
-          git fetch --prune --unshallow
+      - name: Figure out and apply the next canary version
+        run: npx nx run repo-tools:apply-canary-version
 
-      - name: Publish all packages to npm
-        run: npx lerna publish --loglevel=verbose --canary --exact --force-publish --yes
+      - name: Publish all packages to npm with the canary tag
+        # NOTE: this needs to be npx, rather than yarn, to make sure the authenticated npm registry is used
```

Comment:
[Question] It feels like we should be able to get Yarn to work with npm auth, no? `.yarnrc.yml` or some such?

Not a blocker IMO, just a curiosity.

Reply:
npx seems to be consistently used in the workflow already anyway, this comment was carried over from another repo where I had that experience but it was a while ago so fuzzy on the details